### PR TITLE
Release Astro Certified 1.10.7-10

### PR DESCRIPTION
**Pull request by kaxil**
AC Airflow Tag: https://github.com/astronomer/airflow/releases/tag/v1.10.7%2Bastro.10

**Summary**:

Allow Adding & Retrieving Airflow Connections & Variables from different Secret Backends. Backported from Airflow 1.10.10

### New Features

- Allow setting Airflow Variable values to empty string ('') ([commit](https://github.com/astronomer/airflow/commit/d6d666c33)) 
- Get Airflow Variables from AWS Systems Manager Parameter Store ([commit](https://github.com/astronomer/airflow/commit/6a7130f55)) 
- Get Airflow Variables from GCP Secrets Manager ([commit](https://github.com/astronomer/airflow/commit/595ede548)) 
- Get Airflow Variables from Hashicorp Vault ([commit](https://github.com/astronomer/airflow/commit/59435bb88)) 
- Get Airflow Variables from Environment Variables ([commit](https://github.com/astronomer/airflow/commit/fe1a6a3bd)) 
- [AIRFLOW-6987] Avoid creating default connections ([commit](https://github.com/astronomer/airflow/commit/89052690e)) 
- Make BaseSecretsBackend.build_path generic ([commit](https://github.com/astronomer/airflow/commit/1623b80bc)) 
- Fix CloudSecretsManagerBackend invalid connections_prefix ([commit](https://github.com/astronomer/airflow/commit/fe0a9d850)) 
- Standardize SecretBackend class names ([commit](https://github.com/astronomer/airflow/commit/008cb6ebe)) 
- [AIRFLOW-7105] Unify Secrets Backend method interfaces ([commit](https://github.com/astronomer/airflow/commit/5ddff01e8)) 
- [AIRFLOW-7104] Add Secret backend for GCP Secrets Manager ([commit](https://github.com/astronomer/airflow/commit/19bcf781b)) 
- [AIRFLOW-5705] Make AwsSsmSecretsBackend consistent with VaultBackend ([commit](https://github.com/astronomer/airflow/commit/cff559cbb)) 
- [AIRFLOW-7076] Add support for HashiCorp Vault as Secrets Backend ([commit](https://github.com/astronomer/airflow/commit/99bb2a7ef)) 
- [AIRFLOW-5705] Fix bugs in AWS SSM Secrets Backend ([commit](https://github.com/astronomer/airflow/commit/afae4b62b)) 
- [AIRFLOW-5705] Fix bug in Secrets Backend ([commit](https://github.com/astronomer/airflow/commit/29015f537)) 
- [AIRFLOW-5705] Add secrets backend and support for AWS SSM ([commit](https://github.com/astronomer/airflow/commit/2a82f5365)) 

### Improvements

- [AIRFLOW-5167] Update dependencies for GCP packages ([commit](https://github.com/astronomer/airflow/commit/6277fcd499))

### CODE REVIEW

**Comment by kaxil** [comment]
I have tested `astronomerio/ap-airflow:1.10.7-10-buster-onbuild` image on Prod deployment. 

Works fine, the version check plugin does not error either. I will create a separate PR to add a constraint on the Version check plugin (although it is not strictly required)

**Comment by kaxil** [comment]
Converting this to DRAFT as "Example Connections" are not shown in Airflow 1.10.7-10 . Checking it

**Comment by kaxil** [comment]
This is ready for review again. @pritt20 and I both have tested this on Staging & Prod.

The default connections don't load because we run `airflow upgradedb` and not `airflow initdb`.

File: 1.10.10/buster/Dockerfile [hunk 1]
```diff
@@ -113,7 +113,7 @@ COPY include/pip.conf /etc/pip.conf
 COPY include/pip-constraints.txt /usr/local/share/astronomer-pip-constraints.txt
 
 # Pip install airflow and astro security manager
-RUN pip install "${AIRFLOW_MODULE}" --pre \
+RUN pip install --no-cache-dir "${AIRFLOW_MODULE}" \
```
Comment (ashb) [inline 1.10.10/buster/Dockerfile:116]: This option not needed as line 91 sets `ENV PIP_NO_CACHE_DIR="true"`

File: 1.10.7/buster/Dockerfile [hunk 2]
```diff
@@ -115,7 +115,7 @@ COPY include/pip.conf /etc/pip.conf
 COPY include/pip-constraints.txt /usr/local/share/astronomer-pip-constraints.txt
 
 # Pip install airflow and astro security manager
-RUN pip install "${AIRFLOW_MODULE}" \
+RUN pip install --no-cache-dir "${AIRFLOW_MODULE}" \
```
Comment (ashb) [inline 1.10.7/buster/Dockerfile:118]:
```suggestion
RUN pip install "${AIRFLOW_MODULE}" \
```

same here

File: 1.10.10/buster/Dockerfile (hunk 1 above)
Comment (ashb) [inline 1.10.10/buster/Dockerfile:116]:
```suggestion
RUN pip install "${AIRFLOW_MODULE}" \
```

**Pull request merged**